Firestore/core/src/firebase/firestore/nanopb/writer.cc

```diff
@@ -54,6 +54,23 @@ Writer Writer::Wrap(std::vector<uint8_t>* out_bytes) {
   return Writer(raw_stream);
 }
 
+Writer Writer::Wrap(std::string* out_string) {
```

Comment:
Optional: this implementation looks exactly the same as the one for `vector`, except for the output type, so perhaps it could be delegated to a template function (or even the public method could be a template).

Reply:
Done, though only via a private template function. The extra flexibility that comes from allowing arbitrary containers here is unlikely to prove useful and could be a source of bugs because the types seem unconstrained by the implementation. For example, the following is allowed by the template:
```c++
Writer Writer::Wrap(std::vector<int64_t>* out_bytes) {
  return Writer{WrapContainer(out_bytes)};
}
```

Given the low-level nature of what we're trying to do here I think it's reasonable for the factory functions to have concrete types.

Reply:
Thanks! (I guess it would be possible to `static_assert` the size of the element, but I agree it goes into overkill territory)